sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -480,7 +480,7 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
     try {
       Option(hive.getFunction(db, name)).map(fromHiveFunction)
     } catch {
-      case CausedBy(ex: NoSuchObjectException) if ex.getMessage.contains(name) =>
+      case CausedBy(ex: Exception) if ex.getMessage.contains(s"$name does not exist") =>
```

Comment:
The objective here is not to catch all the exceptions but the ones caused by the function not existing. In my case, this exception is "org.apache.hadoop.hive.ql.metadata.HiveException: MetaException(message:NoSuchObjectException(message:Function default.func does not exist))" whose root cause is MetaException, but it may vary in different situations (not really sure it varies, just conjecture based on previous code. See pr #12198 and #12853).